types/ckeditor__ckeditor5-core/src/editor/editorui.d.ts

```diff
@@ -2,21 +2,108 @@ import ComponentFactory from '@ckeditor/ckeditor5-ui/src/componentfactory';
 import { Emitter, EmitterMixinDelegateChain } from '@ckeditor/ckeditor5-utils/src/emittermixin';
 import EventInfo from '@ckeditor/ckeditor5-utils/src/eventinfo';
 import FocusTracker from '@ckeditor/ckeditor5-utils/src/focustracker';
+import { BindChain, Observable } from '@ckeditor/ckeditor5-utils/src/observablemixin';
 import { PriorityString } from '@ckeditor/ckeditor5-utils/src/priorities';
 import Editor from './editor';
 
-export default class EditorUI implements Emitter {
+/**
+ * A class providing the minimal interface that is required to successfully bootstrap any editor UI.
+ */
+export default class EditorUI implements Observable {
+    /**
+     * Creates an instance of the editor UI class.
+     */
     constructor(editor: Editor);
+
+    /**
+     * The editor that the UI belongs to.
+     */
     readonly editor: Editor;
+
+    /**
+     * An instance of the {@link module:ui/componentfactory~ComponentFactory}, a registry used by plugins
+     * to register factories of specific UI components.
+     */
     readonly componentFactory: ComponentFactory;
+
+    /**
+     * Stores the information about the editor UI focus and propagates it so various plugins and components
+     * are unified as a focus group.
+     */
     readonly focusTracker: FocusTracker;
-    get viewportOffset(): { top: number };
-    protected set viewportOffset(value: { top: number });
-    readonly element: null;
+
+    /**
+     * Stores viewport offsets from every direction.
+     *
+     * Viewport offset can be used to constrain balloons or other UI elements into an element smaller than the viewport.
+     * This can be useful if there are any other absolutely positioned elements that may interfere with editor UI.
+     *
+     * Example `editor.ui.viewportOffset` returns:
+     *
+     * ```js
+     * {
+     * 	top: 50,
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/ckeditor__ckeditor5-core/src/editor/editorui.d.ts;
  Line: 45;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.